internal/cri/server/container_stats_list.go

```diff
@@ -51,14 +62,20 @@ func (c *criService) ListContainerStats(
 	if err != nil {
 		return nil, fmt.Errorf("failed to fetch metrics for tasks: %w", err)
 	}
-	criStats, err := c.toCRIContainerStats(ctx, resp.Metrics, containers)
+	css, err := c.toContainerStats(ctx, resp.Metrics, containers)
 	if err != nil {
 		return nil, fmt.Errorf("failed to convert to cri containerd stats format: %w", err)
 	}
-	return criStats, nil
+	return css, nil
 }
 
-type metricsHandler func(containerstore.Metadata, *types.Metric) (*runtime.ContainerStats, error)
+type containerStats struct {
+	stats *runtime.ContainerStats
+	// pids is only valid in linux platform
+	pids uint64
```

Comment:
Wondering if this would be a potential candidate for inclusion in `runtime.ContainerStats`, so that we don't need to define a wrapper in future 🤔

Reply:
I was tempted  to do so as well, but pids doesn't seem a common field for all platforms (i.e. windows) so not sure how feasible it is.

Reply:
Yes, I was curious about the Windows case; maybe there's still an equivalent somewhere, but if not, then it _could_ possibly still be fine as long as it's documented "not used on Windows platforms".

@kevpar @kiashok do you know if there's an equivalent on Windows that could be implemented in future (and/or if it's relevant for Windows?)

Reply:
FWIW; I don't think the current wrapper is too problematic; it's not exported, so only used internally. I merely went looking if we could reduce code-churn somehow, but as the `runtime.ContainerStats` is not owned by containerd, that would require upstream changes first, so in the meantime having the wrapper isn't a huge issue (for me at least 😄)

Reply:
Just to make sure I'm understanding: `pids` is the number of processes in the container, and the question is "does Windows support querying for this as well, so that it would make sense to make it a common field?"?

HCS does support querying for an array of info on each process in the container, so you could use that to determine the count. At the HCS level it is returned as a JSON object with an array of process entries, rather than a single count, though.

Reply:
IMO, we can create new issue to track this question. This patch is to reduce api calls. It's not intend to introduce new field in CRI-API.

Reply:
Thanks @kevpar yes, I think "yes to both" - if having this information makes sense to both Windows and Linux, possibly it could be included in the type.

Definitely not a blocker (as mentioned; changes in this PR are internal, so perfectly fine for now).